src/callbacks_step/save_solution.jl

```diff
@@ -52,25 +64,69 @@ function Base.show(io::IO, ::MIME"text/plain", cb::DiscreteCallback{<:Any, <:Sav
   end
 end
 
+function Base.show(io::IO, ::MIME"text/plain",
+                   cb::DiscreteCallback{<:Any, <:PeriodicCallbackAffect{<:SaveSolutionCallback}})
+  @nospecialize cb # reduce precompilation time
 
-function SaveSolutionCallback(; interval=0,
+  if get(io, :compact, false)
+    show(io, cb)
+  else
+    save_solution_callback = cb.affect!.affect!
+
+    setup = [
+             "dt" => save_solution_callback.interval_or_dt,
+             "solution variables" => save_solution_callback.solution_variables,
+             "save initial solution" => save_solution_callback.save_initial_solution ? "yes" : "no",
+             "save final solution" => save_solution_callback.save_final_solution ? "yes" : "no",
+             "output directory" => abspath(normpath(save_solution_callback.output_directory)),
+            ]
+    summary_box(io, "SaveSolutionCallback", setup)
+  end
+end
+
+
+function SaveSolutionCallback(; interval::Integer=0,
+                                dt=nothing,
                                 save_initial_solution=true,
                                 save_final_solution=true,
                                 output_directory="out",
                                 solution_variables=cons2prim)
 
-  solution_callback = SaveSolutionCallback(interval, save_initial_solution, save_final_solution,
+  if !isnothing(dt) && interval == 0
+    interval_or_dt = dt
+  else # isnothing(dt) && interval >= 0
+    interval_or_dt = interval
+  end
```

Comment:
Combined suggestion from the earlier review cycle

```suggestion
  if !isnothing(dt) && interval > 0
    throw(ArgumentError("You can either set the number of steps between output (using `interval`) or the time between outputs (using `dt`) but not both simultaneously"))
  end
  
  
  if !isnothing(dt)
    interval_or_dt = dt
  else # isnothing(dt)
    interval_or_dt = interval
  end
```

Reply:
Done